test/test-test.js

```diff
@@ -218,7 +219,24 @@ buster.testCase("sinon-test", {
             callback();
         }).call({}, "arg1", {}, done);
     },
-
+    "promises": {
```

Comment:
It tests that if the wrapped test returns a promise, then the spies are not removed until after the promise resolves, so that asynchronous test code can execute with the spies still in place. (Without the PR, the spies are removed synchronously when a promise is returned.)

I wrapped the test in "promises" because ideally there would also be a test for "cleanup on rejection". However I had trouble with promises, buster & browserify so didn't know how to fix. (I use node or ember & testem if I'm writing for a browser.)